### PR TITLE
Fixes #34782 - Add reports tab in host page

### PR DESCRIPTION
**Pull request by Ron-Lavi**
### Reports Tab
Screenshot (81)

### CODE REVIEW

File: webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/ReportsTable.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    emptyState = (
+      <Grid>
+        <EmptyState>
+          <EmptyStateIcon
+            icon={
+              <ExclamationCircleIcon color="var(--pf-global--palette--red-200)" />
+            }
+          />
+          <Title size="lg" headingLevel="h4">
+            {__('Something went wrong')}
```
Comment (MariaAga) [inline webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/ReportsTable.js]: There should be a detailed error about what went wrong

File: webpack/assets/javascripts/react_app/components/HostDetails/Tabs/index.js [hunk 2]
```diff
@@ -3,13 +3,29 @@ import { translate as __ } from '../../../common/I18n';
 import { addGlobalFill } from '../../common/Fill/GlobalFill';
 import { DEFAULT_TAB, TABS_SLOT_ID } from '../consts';
 import OverviewTab from './Overview';
+import ReportsTab from './ReportsTab';
+
+const hostPageTabs = [
```
Comment (MariaAga) [inline webpack/assets/javascripts/react_app/components/HostDetails/Tabs/index.js]: This should probably be the same as in here, I like both approaches
https://github.com/theforeman/foreman/pull/9189/files#diff-e3d78a4ebd8f436480215104bff5ebc112ca0aef73dc230d08711daa2c60ad4cR8

pr:https://github.com/theforeman/foreman/pull/9189

File: webpack/assets/javascripts/react_app/components/HostDetails/Tabs/index.js [hunk 3]
```diff
@@ -3,13 +3,29 @@ import { translate as __ } from '../../../common/I18n';
...
+];
 
 export const registerCoreTabs = () => {
-  addGlobalFill(
-    TABS_SLOT_ID,
-    DEFAULT_TAB,
-    <OverviewTab key="host-details-overview-tab" />,
-    1000,
-    { title: __('Overview') }
+  hostPageTabs.forEach(({ name, Component, weight }) =>
```
Comment (MariaAga) [inline webpack/assets/javascripts/react_app/components/HostDetails/Tabs/index.js]: Should include `options`

File: webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/index.js [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+      const {
+        page: urlPage,
+        perPage: urlPerPage,
+        search: urlSearch,
+      } = getUrlParams();
+      const search = searchParam !== undefined ? searchParam : urlSearch;
+      const page = pageParam || urlPage;
+      const per_page = perPageParam || urlPerPage;
+      dispatch(
+        get({
```
Comment (MariaAga) [inline webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/index.js]: Will using `useAPI` make this shorter?

File: webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/index.js [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+      params.perPage = Number(urlParams.get('per_page')) || params.perPage;
+    }
+    return params;
+  }, [history.location, settingsPerPage]);
+
+  const updateUrl = useCallback(
+    ({ page, per_page, search = '' }) => {
+      const uri = new URI();
+      uri.search({ page, per_page, search });
+      history.push({ search: uri.search() });
```
Comment (MariaAga) [inline webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/index.js:97]: Isn't this the default behavior from the pagination component?

File: webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/index.js [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+          data={{
+            ...getControllerSearchProps('/config_reports'),
+            controller: 'config_reports',
+          }}
+          onSearch={search => fetchReports({ search, page: 1 })}
+          initialQuery={getUrlParams().search}
+        />
+      </GridItem>
+      <GridItem span={4}>
+        <Divider isVertical />
```
Comment (MariaAga) [inline webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/index.js]: Can there be some kind of documentation button or explanation tooltip to what this page is?

File: webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/index.js [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+
+  return (
+    <Grid id="new_host_details_insights_tab" hasGutter style={{ padding: 24 }}>
+      <GridItem span={5}>
+        <SearchBar
+          data={{
+            ...getControllerSearchProps('/config_reports'),
+            controller: 'config_reports',
+          }}
+          onSearch={search => fetchReports({ search, page: 1 })}
```
Comment (MariaAga) [inline webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/index.js:110]: `OnBookmarkClick` should have the same function

File: webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/helpers.js [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+    );
+  if (restarted)
+    summary.push(
+      <IconWrapper key="restarted" label={restarted} title={__('Restarted')}>
+        <SyncAltIcon color="var(--pf-global--palette--orange-300)" />
+      </IconWrapper>
+    );
+  if (applied)
+    summary.push(
+      <IconWrapper key="applied" label={applied} title={__('applied')}>
```
Comment (MariaAga) [inline webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/helpers.js]:
```suggestion
      <IconWrapper key="applied" label={applied} title={__('Applied')}>
```

**Review by MariaAga — CHANGES_REQUESTED** [review]
Thanks Ron! I still need to test it, but for now I left some comments on the code itself

File: app/controllers/config_reports_controller.rb [hunk 9]
```diff
@@ -49,4 +68,19 @@ def csv_columns
...
+  def can_view?(report)
+    authorized_for(auth_object: report, authorizer: authorizer, permission: "view_#{controller_permission}")
+  end
+
+  def origin_icon(report)
+    origin = report.origin
+    return 'N/A' if origin.blank?
+    origin_icon = try("#{origin.downcase}_report_origin_icon".to_sym)
+    helpers.image_tag(origin_icon || origin + ".png", :title => _("Reported by %s") % origin, :width => "20px")
+  end
```
Comment (MariaAga) [inline app/controllers/config_reports_controller.rb]: Can the icon building be done in the UI instead and here only send a value `null/ansible/...`

File: webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/index.js (hunk 4 above)
Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/index.js]: well, I need to also pass the `fetchReports` as a callback around, so I guess it's better to use it this way

File: webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/index.js (hunk 5 above)
Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/index.js:97]: yes, but not with the hash router :)

File: webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/index.js (hunk 6 above)
Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/index.js]: as a follow up we will add more buttons and quick filters around

File: webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/ReportsTable.js (hunk 1 above)
Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/ReportsTable.js]: Would it be OK to handle that in a follow-up?

File: webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/helpers.js [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+  PendingIcon,
+  AngleDoubleRightIcon,
+} from '@patternfly/react-icons';
+import { openConfirmModal } from '../../../ConfirmModal';
+import { APIActions } from '../../../../redux/API';
+import { translate as __ } from '../../../../common/I18n';
+import RelativeDateTime from '../../../common/dates/RelativeDateTime';
+import PuppetPng from '../../../../../../images/Puppet.png';
+import AnsiblePng from '../../../../../../images/Ansible.png';
+
```
Comment (MariaAga) [inline webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/helpers.js]: Can these logos be reused from `app/assets/images/`?

File: webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/ReportsTable.js (hunk 1 above)
Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/ReportsTable.js]: Yeah, can you open a redmine?

File: webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/helpers.js (hunk 10 above)
Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/helpers.js]: done :)

File: webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/ReportsTable.js (hunk 1 above)
Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/HostDetails/Tabs/ReportsTab/ReportsTable.js]: never mind, added in this PR and also reused Foreman's EmptyState

**Review by MariaAga — APPROVED** [review]
LGTM!

**Comment by MariaAga** [comment]
merged, thanks @Ron-Lavi!

**Pull request merged**